aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1550,18 +1551,20 @@ inline void baddbmm_cpu_kernel(const Tensor& result, const Tensor& self, const T
           auto r2 = r1[i];
           auto s2 = s1[i];
           for (const auto j : c10::irange(js)) {
-            scalar_t &r = r2[j];
+            opmath_t acc_value = r2[j];
+            scalar_t& r = r2[j];
             if (is_bmm) {
-              r = 0;
+              acc_value = 0;
               for (const auto k : c10::irange(ks)) {
-                r += s2[k] * m1[k][j];
+                acc_value += s2[k] * m1[k][j];
               }
             } else {
-              r *= beta;
+              acc_value *= beta;
               for (const auto k : c10::irange(ks)) {
-                r += alpha * s2[k] * m1[k][j];
+                acc_value += alpha * s2[k] * m1[k][j];
               }
             }
+            r = acc_value;
```

Comment:
Thanks for advice, I followed this with a little change since for baddbmm `acc_value` should got from `r2[j]`
```
opmath_t acc_value = is_bmm ? opmath_t(0) : opmath_t(r2[j]);
```